src/xenia/cpu/backend/x64/x64_sequences.cc

```diff
@@ -1774,10 +1897,28 @@ struct DIV_I32 : Sequence<DIV_I32, I<OPCODE_DIV, I32Op, I32Op, I32Op>> {
       assert_true(!i.src1.is_constant);
       e.mov(e.ecx, i.src2.constant());
       if (i.instr->flags & ARITHMETIC_UNSIGNED) {
-        e.mov(e.eax, i.src1);
-        // Zero upper bits.
-        e.xor_(e.edx, e.edx);
-        e.div(e.ecx);
+        auto magicgu_val = magicu<unsigned>(i.src2.value->constant.u32);
+
+        if (!magicgu_val.a) {
+          e.mov(i.dest, i.src1);
+          if (e.IsFeatureEnabled(kX64EmitAVX2)) {
+            e.mov(e.edx, magicgu_val.M);
+            e.mulx(i.dest, i.dest, i.dest);
+          } else {
+            e.mov(e.eax, magicgu_val.M);
+            e.mul(i.dest);
+            e.mov(i.dest, e.edx);
+          }
+          if (magicgu_val.s != 0) e.shr(i.dest, magicgu_val.s);
```

Comment:
Put this in a proper code block. One liners like this are annoying to read and maintain.